python/tests/scripts/d5_auto_types.metta

```diff
@@ -1,49 +1,61 @@
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+Auto type-checking can be enabled
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 !(pragma! type-check auto)
 
 ; checking grounded types
 !(assertEqualToResult
    (+ 5 "S")
   ((Error "S" BadType)))
-
 !(assertEqualToResult
    (== 5 "S")
   ((Error "S" BadType)))
 
-; checking custom types
+; declaring some custom types
 (: Entity Type)
 (: Socrates Entity)
 (: Human (-> Entity Type))
 (: Mortal (-> Entity Type))
 (: HumansAreMortal (-> (Human $t) (Mortal $t)))
 (: SocratesIsHuman (Human Socrates))
 (: SamIsMortal (Mortal Sam))
+
 ; well typed
 (HumansAreMortal SocratesIsHuman)
 (== Mortal Human)
-; badly typed
+
+; badly typed examples produce `Error` grounded values describing the issue:
 !(assertEqualToResult
    (== SocratesIsHuman SamIsMortal)
   ((Error SamIsMortal BadType)))
 !(assertEqualToResult
    (HumansAreMortal (Human Socrates))
   ((Error (Human Socrates) BadType)))
 
-
-; we'll got the error, although `collapse` will additionally
-; wrap it into expression
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; `collapse` evaluates its argument, so it shows a type error
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 !(assertEqualToResult
    (collapse (+ 5 "S"))
    (((Error "S" BadType))))
 
-; `match` can accept badly typed expressions without type error,
-; which is correct
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; match does not perform type checking on its pattern,
+; so it accepts a badly typed expression, which is expected
```

Comment:
We can add that this allows us to write code that analysis badly typed expressions in MeTTa itself (e.g. for self-reflective genetic programming, or whatever). Basically, it might be useful to underline that there are cases when we do want to deal with "wrong" MeTTa programs on a meta-level in MeTTa itself.

Reply:
new proposal:
```
; `match` does not perform type checking on its pattern,
; so it accepts a badly typed expression, which is expected.
; It can be useful to deal with "wrong" MeTTa programs on
; a meta-level in MeTTa itself, so this behavior of `match`
; allows us to write code that analyzes badly typed expressions
; within MeTTa (e.g. for self-reflective genetic programming).
```